### PR TITLE
Common dev mode launcher API for Maven and Gradle

### PR DESCRIPTION
**Pull request by aloubyansky**
Originally I was looking into adding support for watching the Gradle build scripts (as we do for the POM files) in dev mode but stumbled on process management in Gradle. However, I thought until I figure that out this common dev mode launcher helper may already make sense on its own.

### CODE REVIEW

File: devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java [hunk 1]
```diff
@@ -194,205 +180,113 @@ public void startDev() {
...
-                manifest.write(out);
-
-                out.putNextEntry(new ZipEntry(DevModeMain.DEV_MODE_CONTEXT));
-                ByteArrayOutputStream bytes = new ByteArrayOutputStream();
-                try (ObjectOutputStream obj = new ObjectOutputStream(new DataOutputStream(bytes))) {
-                    obj.writeObject(context);
-                }
-                out.write(bytes.toByteArray());
+        //we also want to add the maven plugin jar to the class path
+        //this allows us to just directly use classes, without messing around copying them
```
Comment (glefloch) [inline devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java:252]: Should be Gradle here no?

**Review by glefloch — APPROVED** [review]
Nice work! I also tried to restart devmode on build script change but I was not able to make Gradle recompute it's configuration using the public API.

File: devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java (hunk 1 above)
Comment (aloubyansky) [reply devtools/gradle/src/main/java/io/quarkus/gradle/tasks/QuarkusDev.java:252]: Ah, right, thanks. It's apparently been there before, I'll fix it.

**Comment by aloubyansky** [comment]
> Nice work! I also tried to restart devmode on build script change but I was not able to make Gradle recompute it's configuration using the public API.

Did you use the same way to resolve the AppModel? i.e. `extension().getAppModelResolver(LaunchMode.DEVELOPMENT)`? We shouldn't be using the current project for that. We should be using the connect and custom model loading api instead. Or was that what you tried?

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**